pulsar-broker/src/main/java/org/apache/pulsar/broker/offload/impl/BlobStoreBackedInputStreamImpl.java

```diff
@@ -76,26 +73,27 @@ private boolean refillBufferIfNeeded() throws IOException {
             long startRange = cursor;
             long endRange = Math.min(cursor + bufferSize - 1,
                                      objectLen - 1);
-            GetObjectRequest req = new GetObjectRequest(bucket, key)
-                .withRange(startRange, endRange);
-            log.debug("Reading range {}-{} from {}/{}", startRange, endRange, bucket, key);
-            try (S3Object obj = s3client.getObject(req)) {
-                versionCheck.check(key, obj.getObjectMetadata());
-
-                Long[] range = obj.getObjectMetadata().getContentRange();
-                long bytesRead = range[1] - range[0] + 1;
 
-                buffer.clear();
-                bufferOffsetStart = range[0];
-                bufferOffsetEnd = range[1];
-                InputStream s = obj.getObjectContent();
-                int bytesToCopy = (int)bytesRead;
-                while (bytesToCopy > 0) {
-                    bytesToCopy -= buffer.writeBytes(s, bytesToCopy);
+            try {
+                Blob blob = blobStore.getBlob(bucket, key);
+                versionCheck.check(key, blob);
+                PayloadSlicer slicer = new BasePayloadSlicer();
```

Comment:
Slicer isnt what you need here. getBlob can take a GetOptions parameter. You can set the range in that.